### PR TITLE
Make Error types `Hashable` + `Equatable`

### PR DESCRIPTION
**Pull request by felixschlegel**
### Modifications:

* `KafkaError`: make `Equatable`
* `KafkaAcknowledgedMessageError`: make `Equatable` + `Hashable`

### CODE REVIEW

File: Sources/SwiftKafka/KafkaAcknowledgedMessageError.swift [hunk 1]
```diff
@@ -62,3 +62,11 @@ public struct KafkaAcknowledgedMessageError: Error, CustomStringConvertible {
...
     }
 }
+
+// MARK: - KafkaAcknowledgedMessageError + Hashable
+
+extension KafkaAcknowledgedMessageError: Hashable {}
+
+// MARK: - KafkaAcknowledgedMessageError + Equatable
+
+extension KafkaAcknowledgedMessageError: Equatable {}
```
Comment (FranzBusch) [inline Sources/SwiftKafka/KafkaAcknowledgedMessageError.swift:72]: `Hashable` implies `Equatable`

File: Sources/SwiftKafka/KafkaError.swift [hunk 2]
```diff
@@ -228,3 +230,11 @@ extension KafkaError {
...
     }
 }
+
+// MARK: - KafkaError + Hashable
+
+extension KafkaError: Hashable {}
+
+// MARK: - KafkaError + Equatable
+
+extension KafkaError: Equatable {}
```
Comment (FranzBusch) [inline Sources/SwiftKafka/KafkaError.swift:240]: Same here.

File: Sources/SwiftKafka/KafkaAcknowledgedMessageError.swift (hunk 1 above)
Comment (felixschlegel) [reply Sources/SwiftKafka/KafkaAcknowledgedMessageError.swift]: Thanks for pointing that out, also removed redundant annotations of `Equatable` for other types that we have

**Review by FranzBusch — APPROVED** [review]

**Comment by FranzBusch** [comment]
@felixschlegel can you fix the soundness failure

**Comment by felixschlegel** [comment]
> @felixschlegel can you fix the soundness failure

This is currently blocked by a [CI issue](https://ci.swiftserver.group/job/swift-kafka-gsoc-soundness-prb/88/console):

```
Creating network "swift-kafka-gsoc-soundness-prb_default" with the default driver
could not find an available, non-overlapping IPv4 address pool among the defaults to assign to the network
Build step 'Execute shell' marked build as failure
```

**Pull request merged**